Old slab version in dependencies breaks build

async-executor has a dependency on a really old version of slab.

`slab = "0.4.4"`

The problem is that this version no longer compiles with any remotely recent version of Rust:

```
error[E0015]: cannot call non-const fn `Slab::<Waker>::new` in constant functions
   --> /home/robin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-executor-1.13.1/src/lib.rs:693:32
    |
693 |             active: Mutex::new(Slab::new()),
    |                                ^^^^^^^^^^^
    |
    = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
```

This was fixed back in 2022 and shipped with slab 0.4.7.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,7 @@ async-task = "4.4.0"
 concurrent-queue = "2.5.0"
 fastrand = "2.0.0"
 futures-lite = { version = "2.0.0", default-features = false }
-slab = "0.4.4"
+slab = "0.4.7"
 
 [target.'cfg(target_family = "wasm")'.dependencies]
 futures-lite = { version = "2.0.0", default-features = false, features = ["std"] }

